backend/storage-service/src/jvmMain/kotlin/file/services/SynchronizationService.kt

```diff
@@ -0,0 +1,205 @@
+                        path,
+                        user_id,
+                        access_type
+                    ) values (
+                        :id,
+                        :device,
+                        :path,
+                        :user,
+                        :access
+                    )
+                """
+            )
+        }
+
+        syncthing.writeConfig()
+    }
+
+    suspend fun removeFolder(actor: Actor, request: SynchronizationRemoveFolderRequest) {
+        db.withSession { session ->
+            session.sendPreparedStatement(
+                {
+                    setParameter("id", request.id)
+                    setParameter("user", actor.username)
+                },
+                """
+                    delete from storage.synchronized_folders
+                    where id = :id and user_id = :user
+                """
+            )
+        }
+
+        syncthing.writeConfig()
+    }
+
+    suspend fun addDevice(actor: Actor, request: SynchronizationAddDeviceRequest) {
+        if (syncthing.config.devices.find { it.id == request.id } != null) {
+            throw RPCException.fromStatusCode(HttpStatusCode.BadRequest)
+        }
+
+        db.withSession { session ->
+            session.sendPreparedStatement(
+                {
+                    setParameter("device", request.id)
+                    setParameter("user", actor.username)
+                },
+                """
+                    insert into storage.user_devices(
+                        device_id,
+                        user_id
+                    ) values (
+                        :device,
+                        :user
+                    )
+                """
+            )
+        }
+
+        syncthing.writeConfig()
+    }
+
+    suspend fun removeDevice(actor: Actor, request: SynchronizationRemoveDeviceRequest) {
+        db.withSession { session ->
+            session.sendPreparedStatement(
+                {
+                    setParameter("id", request.id)
+                    setParameter("user", actor.username)
+                },
+                """
+                    delete from storage.user_devices
+                    where device_id = :id and user_id = :user
+                """
+            )
+        }
+
+        syncthing.writeConfig()
+    }
+
+    suspend fun browseDevices(actor: Actor): PageV2<SynchronizationDevice> {
+        val devices = db.withSession { session ->
+            session.sendPreparedStatement(
+                {
+                    setParameter("user", actor.username)
+                },
+                """
+                        select device_id
+                        from storage.user_devices
+                        where user_id = :user
+                        limit 100
+                    """
+            ).rows.map { SynchronizationDevice(it.getField(UserDevicesTable.device)) }
+        }
+
+        return PageV2(100, devices, null)
```

Comment:
This code doesn't actually paginate. Instead it returns 100 random user devices. It is random because you don't do any ordering of the devices.

You should look at the `db.paginateV2` call.